test/test_segment_reductions.py

```diff
@@ -11,28 +11,40 @@
 
 
 class TestSegmentReductions(TestCase):
-    def _test_max_simple_1d(self, device, dtype, unsafe):
+    def _test_max_simple_1d(self, device, dtype, unsafe, axis):
         lengths = torch.tensor([1, 2, 3], device=device)
-        data = torch.tensor([1, float("nan"), 3, 4, 5, 6], device=device, dtype=dtype)
-        expected_result = torch.tensor([1, float("nan"), 6], device=device, dtype=dtype)
+        data = torch.tensor(
+            [1, float("nan"), 3, 4, 5, 5],
+            device=device,
+            dtype=dtype,
+            requires_grad=True,
+        )
+        expected_result = torch.tensor([1, float("nan"), 5], device=device, dtype=dtype)
         actual_result = torch.segment_reduce(
-            data=data, reduce="max", lengths=lengths, axis=0, unsafe=unsafe
+            data=data, reduce="max", lengths=lengths, axis=axis, unsafe=unsafe
         )
         self.assertEqual(
             expected_result, actual_result, rtol=1e-03, atol=1e-05, equal_nan=True
         )
-        actual_result = torch.segment_reduce(
-            data=data, reduce="max", lengths=lengths, axis=-1, unsafe=unsafe
-        )
+
+        # Backward is only supported for cpu tensors for now. Return early if cuda
+        if data.is_cuda:
+            return
+
+        # Test backward
+        expected_grad = torch.tensor([1, 1, 0, 0, 0.5, 0.5], device=device, dtype=dtype)
+        actual_result.sum().backward()
         self.assertEqual(
-            expected_result, actual_result, rtol=1e-03, atol=1e-05, equal_nan=True
+            expected_grad, data.grad, rtol=1e-03, atol=1e-05, equal_nan=True
```

Comment:
direct checking of gradients is good, but you should also test with `gradcheck`

Reply:
is there an example I can check for this that works with ops that have key value arguments?